src/view/event-form-view.js

```diff
@@ -156,25 +156,46 @@ function createEventFormTemplate(event, eventTypes) {
   );
 }
 
-export default class EventFormView {
-  constructor({ event, eventTypes }) {
+export default class EventFormView extends AbstractView {
+  #handleSubmit = null;
+  #handleClick = null;
+
+  constructor({ event, eventTypes, onSubmit, onClick }) {
+    super();
     this.event = event;
     this.eventTypes = eventTypes;
+
+    if (onSubmit) {
+      this.#handleSubmit = onSubmit;
+    }
+
+    if (onClick) {
+      this.#handleClick = onClick;
+    }
+
+    this.element.addEventListener('submit', this.#submitHandler);
+
+    const rollupBtnEl = this.element.querySelector('.event__rollup-btn');
+    rollupBtnEl.addEventListener('click', this.#clickHandler);
   }
 
-  getTemplate() {
+  get template() {
     return createEventFormTemplate(this.event, this.eventTypes);
   }
 
-  getElement() {
-    if (!this.element) {
-      this.element = createElement(this.getTemplate());
+  #submitHandler = (evt) => {
+    evt.preventDefault();
+
+    if (this.#handleSubmit) {
+      this.#handleSubmit();
     }
```

Comment:
это не ошибка, но есть более короткий вариант записи 
```suggestion
  
      this.#handleSubmit?.();

```

Reply:
в остальных местах аналогично

Reply:
Круто) С доступом к свойствам объекта в тайпскрипте использую постоянно, но то, что его еще и с функцией можно использовать, не знала. С функцией выглядит причудливо)